packages/core/versioning/core-versioned-http-server/src/types.ts

```diff
@@ -28,52 +26,18 @@ type RqCtx = RequestHandlerContextBase;
  */
 export type Version = `${number}`;
 
-/**
- * Arguments to create a {@link VersionedRouter | versioned router}.
- * @experimental
- */
-export interface CreateVersionedRouterArgs<Ctx extends RqCtx = RqCtx> {
-  /**
-   * A router instance
-   * @experimental
-   */
-  router: IRouter<Ctx>;
-}
-
-/**
- * This interface is the starting point for creating versioned routers and routes
- *
- * @example see ./example.ts
- *
- * @experimental
- */
-export interface VersionHTTPToolkit {
-  /**
-   * Create a versioned router
-   * @param args - The arguments to create a versioned router
-   * @returns A versioned router
-   * @experimental
-   */
-  createVersionedRouter<Ctx extends RqCtx = RqCtx>(
-    args: CreateVersionedRouterArgs<Ctx>
-  ): VersionedRouter<Ctx>;
-}
-
-/**
- * Versioned route access flag, required
- * - '/api/foo' is 'public'
- * - '/internal/my-foo'  is 'internal'
- * Required
- */
-type VersionedRouteConfigOptions = WithRequiredProperty<RouteConfigOptions<RouteMethod>, 'access'>;
 /**
  * Configuration for a versioned route
  * @experimental
  */
 export type VersionedRouteConfig<Method extends RouteMethod> = Omit<
   RouteConfig<unknown, unknown, unknown, Method>,
   'validate' | 'options'
-> & { options: VersionedRouteConfigOptions };
+> & {
+  options?: Omit<RouteConfigOptions<Method>, 'access'>;
+  // Make "access" required and move to top level
+  access: RouteConfigOptions<Method>['access'];
```

Comment:
Moved `access` to top-level config of versioned route.

Reply:
Purely ergonomical so that you don't have to specify `{ options: { access: 'public' } }`, just `{ access: 'public' }`